src/Toml/Bi/Code.hs

```diff
@@ -31,15 +32,26 @@ import qualified Data.Text as Text
 
 -- | Type of exception for converting from 'Toml' to user custom data type.
 data DecodeException
-    = KeyNotFound Key  -- ^ No such key
+    = TrivialError
+    | KeyNotFound Key  -- ^ No such key
     | TableNotFound Key  -- ^ No such table
     | TypeMismatch Key Text TValue  -- ^ Expected type vs actual type
     | ParseError ParseException  -- ^ Exception during parsing
     deriving (Eq, Show)  -- TODO: manual pretty show instances
 
+instance Semigroup DecodeException where
+    TrivialError <> e = e
+    e <> TrivialError = e
```

Comment:
I guess this case is not needed

Reply:
@vrom911 You're right. And removing this case is actually a good improvement, because it will make implementation lazier.